### PR TITLE
Manual merge of sparkMax

### PR DESCRIPTION
**Pull request by worthp**
Manual merge of SparkMax onto feature branch. please review expected changes and merge to master. note SubsystemFactory change to ensure compilation in initFootball() and implementation of moveSideways(double s) as required by DriveTrain.

### CODE REVIEW

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/SubsystemFactory.java [hunk 1]
```diff
@@ -228,6 +230,8 @@ private void initTurbo() throws Exception {
      */
     private void initFootball() throws Exception {
         logger.info("initializing Football");
+        driveTrain  = new SparkMecanum();
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/SubsystemFactory.java]: I just added these lines to make sure both drivetrains compile properly. these can be deleted or moved to to proper sled

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/SubsystemFactory.java [hunk 2]
```diff
@@ -228,6 +230,8 @@ private void initTurbo() throws Exception {
...
+        driveTrain  = new SparkTurboTankDrive();
         // kicker = new Kicker();
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/SubsystemFactory.java]: I just added these lines to make sure both drivetrains compile properly. these can be deleted or moved to to proper sled

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkTurboTankDrive.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+
+public class SparkTurboTankDrive extends SparkDriveTrain {
+
+    private final int turboSolOpenPort = 0;
+    private final int turboSolClosePort = 1;
+
+    private DoubleSolenoid turboSol = new DoubleSolenoid(turboSolOpenPort, turboSolClosePort);
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkTurboTankDrive.java]: Whoa!!!!!! PortMan!!!!! fix this

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkMecanum.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	private double XValScaler1 = 1;
+	private double YValScaler2 = 1;
+	private double XValScaler2 = 1;
+	private double ZValScaler = 1;
+
+	private int velocityInvert1 = 1;
+	private int velocityInvert2 = -1;
+	private int velocityInvert3 = -1;
+	private int velocityInvert4 = 1;
+
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkMecanum.java]: these scalars should be removed. they were an idea from halter last year that was pretty much just dumb. they just add confusion now

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkMecanum.java [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+	private double velocity2;
+	private double velocity3;
+	private double velocity4;
+
+	private double YValScaler1 = 1;
+	private double XValScaler1 = 1;
+	private double YValScaler2 = 1;
+	private double XValScaler2 = 1;
+	private double ZValScaler = 1;
+
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkMecanum.java]: these scalars should be removed. they were an idea from halter last year that was pretty much just dumb. they just add confusion now

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/interfaces/SparkDriveTrain.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+public abstract class SparkDriveTrain extends DriveTrain {
+
+    
+
+    public final Logger logger = Logger.getLogger(LogTest.class.getName());
+
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/interfaces/SparkDriveTrain.java]: this should be SparkDriveTrain.getClass().getName()

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/interfaces/TalonDriveTrain.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+
+public abstract class TalonDriveTrain extends DriveTrain {
+
+    public final Logger logger = Logger.getLogger(LogTest.class.getName());
+
```
Comment (worthp) [inline src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/interfaces/TalonDriveTrain.java]: this should be SparkDriveTrain.getClass().getName()

File: src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkMecanum.java (hunk 5 above)
Comment (Connor2573) [reply src/main/java/org/usfirst/frc/team4611/robot/subsystems/drivetrain/SparkMecanum.java]: agreed

**Pull request merged**